src/app/fdctl/help.c

```diff
@@ -0,0 +1,57 @@
+#include "fdctl.h"
+#include <stdio.h>
+
+#define MAX_LENGTH 4096
+
+#define PRINT_HELP( ... ) do {                                                \
+    int n = snprintf( cur, rem, __VA_ARGS__ );                                \
```

Comment:
Can use `snprintf1` from utility.h which does most of this for you

Reply:
Not super sure how to incorporate the string concatenation when `snprintf1` does not return the number of characters. Can talk offline and incorporate it into another PR if that's okay.